web/ui/react-app/src/App.tsx

```diff
@@ -5,36 +5,62 @@ import { Container } from 'reactstrap';
 import './App.css';
 import { Router, Redirect } from '@reach/router';
 import { Alerts, Config, Flags, Rules, ServiceDiscovery, Status, Targets, TSDBStatus, PanelList } from './pages';
-import PathPrefixProps from './types/PathPrefixProps';
+import { PathPrefixContext } from './contexts/PathPrefixContext';
 
 interface AppProps {
   consolesLink: string | null;
 }
 
-const App: FC<PathPrefixProps & AppProps> = ({ pathPrefix, consolesLink }) => {
+const App: FC<AppProps> = ({ consolesLink }) => {
+  // This dynamically/generically determines the pathPrefix by stripping the first known
+  // endpoint suffix from the window location path. It works out of the box for both direct
+  // hosting and reverse proxy deployments with no additional configurations required.
+  let basePath = window.location.pathname;
+  const paths = [
+    '/graph',
+    '/alerts',
+    '/status',
+    '/tsdb-status',
+    '/flags',
+    '/config',
+    '/rules',
+    '/targets',
+    '/service-discovery',
+  ];
+  if (basePath.endsWith('/')) {
```

Comment:
This condition was necessary to add for when a user navigates to `http://host:3000/` and is redirected to `/graph`; the basePath was resolving to `/` and nav bar links were thus being set to `//graph`, which is breaking.